apis/google/gmail/code/listInbox.py

```diff
@@ -0,0 +1,53 @@
+import os
+
+from googleapiclient.errors import HttpError
+
+from auth import client
+
+
+def main():
+    max_results = os.getenv('MAX_RESULTS')
+    if max_results is not None:
+        max_results = int(max_results)
+    service = client('gmail', 'v1')
+    all_messages = []
+    next_page_token = None
+    try:
+
+        while True:
+            if next_page_token:
+                results = service.users().messages().list(userId='me', labelIds=['INBOX'], pageToken=next_page_token,
+                                                          maxResults=10).execute()
+            else:
+                results = service.users().messages().list(userId='me', labelIds=['INBOX'], maxResults=10).execute()
+            messages = results.get('messages', [])
+            if not messages:
+                break
```

Comment:
Does Python allow you to refer to a variable defined in the `if` statement outside of it? I keep learning new things about this language, lol

Reply:
Yeah they let you do all sorts of crazy things